### DISCUSSION TITLE
DAI 2024-2025 - Java IOs benchmarking - Aladin Iseni

### DISCUSSION THREAD

**Discussion by aladin-heig** [heig-vd-dai-course/heig-vd-dai-course#338] [Show and tell]
https://github.com/aladin-heig/heig-vd-dai-course-java-ios/

## Table for writing files with time in ms
|                 | 1B  | 1KiB | 1MiB | 5MiB |
|-----------------|-----|------|------|------|
| BINARY          | 255 | 340  | 1297 | 5256 |
| BUFFERED_BINARY | 393 | 291  | 200  | 297  |
| TEXT            | 286 | 279  | 365  | 636  |
| BUFFERED_TEXT   | 270 | 286  | 321  | 432  |

## Table for reading files with time in ms
|                 | 1B  | 1KiB | 1MiB | 5MiB |
|-----------------|-----|------|------|------|
| BINARY          | 278 | 266  | 671  | 2238 |
| BUFFERED_BINARY | 260 | 276  | 302  | 446  |
| TEXT            | 262 | 283  | 386  | 630  |
| BUFFERED_TEXT   | 262 | 291  | 320  | 419  |

## Conclusion
For writing and reading small files, time differences are negligeable. But for writing bigger files, the most optimal way is to use the BUFFERED_BINARY function and for reading files it is the BUFFERED_TEXT. The fact that the buffered binary is the fastest for writing is because it uses a buffer in order to optimize the amount of data it sends to the SSD and also because BINARY writes directly the bytes in the data, but TEXT has to interpret as text. In the other hand, for reading fils, both buffered binary and buffered text have almost same performances. Encoding means that the computer has to apply a certain standard for letters in order to be well read and shown to the screen by other softwares.

**Comment by ludelafo** [comment]
Thank you!

You can always find the solution in the solution repository if you ever need to compare your work with the official one here: https://github.com/heig-vd-dai-course/heig-vd-dai-course-solutions.

Do not hesitate to ask questions if something is not clear, either in this discussion or in the Q&A category of the GitHub Discussions here: https://github.com/orgs/heig-vd-dai-course/discussions/categories/q-a.
